src/index.ts

```diff
@@ -7,14 +7,28 @@ interface Heading {
   level: number;
 }
 
+export interface ExtractHeadingsOptions {
```

Comment:
Thank you so much for improving this!

Great work with the JSDoc too. I love the idea of an external config object that'll give devs the flexibility they need.